### PR TITLE
NOISSUE - Create repo structure and workflows

### PR DESCRIPTION
**Pull request by SammyOina**
### What does this do?
Initialize repo strucutre and add workflows

### Which issue(s) does this PR fix/relate to?
None

### List any changes that modify/break current functionality
None

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### CODE REVIEW

**Review by dborovcanin — CHANGES_REQUESTED** [review]
DCO is failing. At least we know checks work. :)

**Comment by SammyOina** [comment]
> DCO is failing. At least we know checks work. :)

all commits now signed, DCO is passing

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        run: go build -v ./...
+
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: --no-config --disable-all --enable gosimple --enable govet --enable unused --enable deadcode --timeout 3m
+
+      - name: Run tests
+        run: go test -mod=vendor -v --race -covermode=atomic -coverprofile cover.out ./...
```
Comment (dborovcanin) [inline .github/workflows/ci.yml]: Add an empty line.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
+# vendor/
+
+# Go workspace file
+go.work
```
Comment (dborovcanin) [inline .gitignore]: Same.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dborovcanin) [reply .github/workflows/ci.yml]: @SammyOina Please do not mark it as resolved but rather actually resolve comments.

File: .gitignore (hunk 2 above)
Comment (dborovcanin) [reply .gitignore]: Same.

File: README.md [hunk 3]
```diff
@@ -1 +1,9 @@
-# migrations
\ No newline at end of file
+# migrations
```
Comment (dborovcanin) [inline README.md]: Start Titles with capital letters.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rodneyosodo) [reply .github/workflows/ci.yml]: Add empty line

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+*.exe~
+*.dll
+*.so
+*.dylib
+# Test binary, built with `go test -c`
+*.test
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+# Go workspace file
+go.work
```
Comment (rodneyosodo) [inline .gitignore]: Add empty line

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions/checkout@v3
+      - name: Build
+        run: go build -v ./...
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: --no-config --disable-all --enable gosimple --enable govet --enable unused --enable deadcode --timeout 3m
+      - name: Run tests
+        run: go test -mod=vendor -v --race -covermode=atomic -coverprofile cover.out ./...
```
Comment (rodneyosodo) [inline .github/workflows/ci.yml]: Add empty line

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**